### PR TITLE
Depend setlocale use to build system provided check

### PR DESCRIPTION
**Pull request by ebraminio**

### CODE REVIEW

File: src/hb-common.cc [hunk 1]
```diff
@@ -35,10 +35,6 @@
 #include <xlocale.h>
 #endif
 
-#ifdef HB_NO_SETLOCALE
-#define setlocale(Category, Locale) "C"
-#endif
```
Comment (ebraminio) [inline src/hb-common.cc]: Apparently redefining this way is not enough, that motivated to look for another solution and thought lets turn it into a check anyway.

**Comment by behdad** [comment]
I thought a lot about what we should check in configure (ie. default disabled) vs define `HB_NO_*` for (ie. default enable).  I prefer that things widely available as part of C standard library to be default-enabled, such that they are picked up automatically by alternative builders.  So, why not keep it that way just change the implementation?

**Comment by behdad** [comment]
Ie. anything that is C89 should be default enabled :).  Maybe more than just that.

**Comment by behdad** [comment]
Probably add `HB_NO_LIBC` or something as well.

**Comment by ebraminio** [comment]
I had to create mock headers also anyway and am no longer seeing this, so :)

**Pull request closed without merging**